cpp/src/arrow/compute/util_internal.h

```diff
@@ -0,0 +1,31 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "arrow/util/logging.h"
+
+namespace arrow {
+namespace util {
+
+template <typename T>
+void CheckAlignment(const void* ptr) {
```

Comment:
Curious why moving this from util.h to a new util_internal.h?

Reply:
this uses a macro that should not end up in the external include files.
util.h ends up in external include files (since we'll need it inside the future libacero)

Reply:
Yes, internal is correct for this, good catch.  `exec/util.h` has always been "implicitly internal" and I've been wanting to rename it forever but hadn't quite gotten around to it.  This is a good start.